### ISSUE TITLE
Implement functionality based on the `old` alias

### ISSUE THREAD

**Issue by ericcornelissen** [simple-icons/simple-icons#5859]
Relates to #4797 and #5627

---

### Description

In Pull Request #5369 we added the possibility for four types of aliases to the JSON data, one of which is `old`. This issue is intended to decide on how we want to use the `old` alias. Feedback and suggestions from anyone are welcome :smile: 

To me the intention of `old` is to establish a clear and easy way to provide backwards compatibility (*1) for brands that have been renamed or slugs that need to be changed for another reason.

### Proposal

With that in mind I propose we implement the following based on the `old` alias: For every brand in [`simple-icons.json`](https://github.com/simple-icons/simple-icons/blob/562aaac0488c17b5288e3018278896dafb2b1735/_data/simple-icons.json) that has an `old` alias, [the build script](https://github.com/simple-icons/simple-icons/blob/562aaac0488c17b5288e3018278896dafb2b1735/scripts/build/package.js) should generate an `.svg` file based on the "original" `.svg` as well as a `.js` file for [the NPM package](https://www.npmjs.com/package/simple-icons) (*2) using the old brand name/slug as file name. The build script should also include the brand using the old brand name/slug in [the NPM package's index.js file](https://github.com/simple-icons/simple-icons/blob/562aaac0488c17b5288e3018278896dafb2b1735/scripts/build/templates/index.js).

### Considerations

Unfortunately this somewhat complicates keeping track of unique slugs. However, as the `old`'s slug was already occupied before the brand renamed, it will never happen that the slug "introduced" by the `old` alias is already in use. That said, we should try to make sure that a new addition that clashes with an `old` alias is detected automatically by tests or linting.

### Open Questions

- (*1) How long should we keep the backwards compatibility? This can be discussed in #5627.
- (*2) Should the SVG `<title>` and JavaScript `title` and `svg` properties have the old brand name or the new brand name. I think it should be the new brand name as the SVG should be the up-to-date version that relates to the new brand name.

**Comment by jorgeamadosoria** [comment]
Can't we just treat the old alias as the aka?

and as for the package, isn't there a way to get a particular svg from different slugs, so you don't have to generate duplicated svgs?

**Comment by ericcornelissen** [comment]
> Can't we just treat the old alias as the aka?

That depends. `aka` doesn't do anything at the moment, so whether or not these can be treated the same depends on what we want either of them to do. To me, the main difference between `aka` and `old` is that `old` is solely to provide backwards compatibility after a brand name has been changed (and therefor has a limited lifetime).

From that perspective, it seems indeed likely that `old` and `aka` have a very similar implementation.


> and as for the package, isn't there a way to get a particular svg from different slugs, so you don't have to generate duplicated svgs?

There isn't just the npm package - there's also the CDNs and packagist package (and to an extend even the npm package) which can all be used to get SVGs directly. For example, if I have the SVG for some brand on my website through the CDN like so

```
<img height="32" width="32" src="https://unpkg.com/simple-icons@v5/icons/branda.svg" />
```

and the brand renames itself, say to

```
<img height="32" width="32" src="https://unpkg.com/simple-icons@v5/icons/brandb.svg" />
```

then to provide backwards compatibility we have to have both `branda.svg` and `brandb.svg` in the published package.

**Comment by jorgeamadosoria** [comment]
> > Can't we just treat the old alias as the aka?
> 
> That depends. `aka` doesn't do anything at the moment, so whether or not these can be treated the same depends on what we want either of them to do. To me, the main difference between `aka` and `old` is that `old` is solely to provide backwards compatibility after a brand name has been changed (and therefor has a limited lifetime).
> 
> From that perspective, it seems indeed likely that `old` and `aka` have a very similar implementation.
> 
> > and as for the package, isn't there a way to get a particular svg from different slugs, so you don't have to generate duplicated svgs?
> 
> There isn't just the npm package - there's also the CDNs and packagist package (and to an extend even the npm package) which can all be used to get SVGs directly. For example, if I have the SVG for some brand on my website through the CDN like so
> 
> ```
> <img height="32" width="32" src="https://unpkg.com/simple-icons@v5/icons/branda.svg" />
> ```
> 
> and the brand renames itself, say to
> 
> ```
> <img height="32" width="32" src="https://unpkg.com/simple-icons@v5/icons/brandb.svg" />
> ```
> 
> then to provide backwards compatibility we have to have both `branda.svg` and `brandb.svg` in the published package.

I understand. That's unfortunate, but I guess there is no other choice but to duplicate the file.
I would keep the old for one or two releases. Are releases happening in set time periods? because if not, I think they should. Like once every three to six months, for example. That should be A LOT of time to give warning.

The aka and old should be added to the text search on the website. Additionally, icons with old aliases should be pointed out somewhere, be it a release notes document (we don't have one of those? we should have one of those,  I think) or in the icon card on the website (I think there is an icon for removal warning of icons in the website repo, so something like that).

I don't know, maybe that's too involved and we should just make old searchable and temporary, and that's it.

**Comment by ericcornelissen** [comment]
> I understand. That's unfortunate, but I guess there is no other choice but to duplicate the file.
I would keep the old for one or two releases. Are releases happening in set time periods? because if not, I think they should. Like once every three to six months, for example. That should be A LOT of time to give warning.

See #5627 for this discussion.


> The aka and old should be added to the text search on the website. Additionally, icons with old aliases should be pointed out somewhere, be it a release notes document (we don't have one of those? we should have one of those, I think) or in the icon card on the website (I think there is an icon for removal warning of icons in the website repo, so something like that).

This is theoretically already possible if we were to use the raw data file in the website build process. The broader question is whether to expose this data in the NPM package as well, but I think that warrants it's own issue (1) to keep the discussion focused and (2) because it's a consideration for every alias.

Until the `old` alias actually does something, I don't think it makes sense to it to any kind of release notes. Even when it does something, I don't think it necessarily makes sense to point it out - but that decision can only be made once we agree on what the `old` alias actually does.


> a release notes document (we don't have one of those? we should have one of those, I think)

There's [the GitHub releases](https://github.com/simple-icons/simple-icons/releases) and the [PRs with the `release` label](https://github.com/simple-icons/simple-icons/pulls?q=is%3Apr+label%3Arelease). So far we have found this to be sufficient. If you think a `CHANGELOG.md` (or similar) should be added, again, feel free to open an issue for that.


> I don't know, maybe that's too involved and we should just make old searchable and temporary, and that's it.

The primary question here is what it means in the context of this repository, not what it means for the website. That is to say, we're not trying to figure out how we want to use the `old` alias on the website here. While that might influence the decision, a solution for the website is not sufficient here as we need to decide what we want to do with the `old` alias in the context of this repository.

**Comment by adamrusted** [comment]
@simple-icons/maintainers - has the 'old' alias been removed? If so, this can be shut down.

**Comment by adamrusted** [comment]
I would suggest, using Twitter/X as an example that we don't necessarily store the old icon - but the alias should provide us with some backward compatibility, so that if/when we remove the icon in a major version - people still using the old slug will receive the new icon (including new title etc) allowing them time to update the slug in their projects.

**Comment by adamrusted** [comment]
Yeah, in the example of clubhouse, I would say if an icon with that exact slug exists, we should still display that, but if no icon with that slug exists, we should search through the aliases. `old` being the top priority.
